### DISCUSSION TITLE
Adding subject name to filename?

### DISCUSSION THREAD

**Discussion by sstaszko** [bonsai-rx/bonsai#1308] [Q&A] [answered]
Hello!

I am using Bonsai as a beginner and was curious if there was a way to add a node (or a better way to program my output filenames) such that I could add a subject's name to all files created in a bonsai session? My workflow is recording and saving video (as avi) as well as video frames timestamps (to csvs) for a Basler camera and two Windows LifeCams. I appreciate any help for incorporating the subject name (as well as if there are obvious improvements to my current workflow). 

![Bonsai Workflow Example](https://user-images.githubusercontent.com/34290980/228014490-3c65b3a5-9629-44cc-a82e-8357d0495a7e.JPG)

**Comment by bruno-f-cruz** [comment, marked as answer]
Hi! 

https://bonsai-rx.org/docs/articles/expressions-format.html?q=Format

Check the documentation of Format for an example on how to do this.

**Reply by sstaszko** [reply]
Thanks so much!
